pyqtgraph/graphicsItems/ViewBox/tests/test_ViewBox.py

```diff
@@ -76,10 +57,62 @@ def test_ViewBox():
     h = vb.geometry().height()
     view1 = QRectF(-5, 0, 20, 10)
     size1 = QRectF(0, h, w, -h)
-    assertMapping(vb, view1, size1)
+    _assert_mapping(vb, view1, size1)
+
+
+def _assert_mapping(vb, r1, r2):
+    assert vb.mapFromView(r1.topLeft()) == r2.topLeft()
+    assert vb.mapFromView(r1.bottomLeft()) == r2.bottomLeft()
+    assert vb.mapFromView(r1.topRight()) == r2.topRight()
+    assert vb.mapFromView(r1.bottomRight()) == r2.bottomRight()
+
+
+
+function_set = set([test_resize, test_wide_resize, test_tall_resize,
+                    test_aspect_ratio_constraint])
+                    
+@pytest.mark.parametrize('function', function_set)
+def setup_function(function):
+    print('\nsetting up function %s' % function)
+    global app, win, vb, QRectF
+    app = pg.mkQApp()
```

Comment:
Even if it wasn't, the "official" rules for QApplication are that you may only create one per process, and it may only be exited once.